services/oauth/v2/destination_info.go

```diff
@@ -0,0 +1,41 @@
+package v2
+
+import "github.com/rudderlabs/rudder-server/utils/misc"
+
+type DestinationInfo struct {
+	WorkspaceID   string
+	DestDefName   string
+	DestDefConfig map[string]interface{}
+	DestinationId string
+	DestConfig    map[string]interface{}
+}
+
+func (d *DestinationInfo) IsOAuthDestination() bool {
+	authValue, err := misc.NestedMapLookup(d.DestDefConfig, "auth", "type")
+	if err != nil {
+		return false
+	}
+	if authType, ok := authValue.(string); ok {
+		return authType == "OAuth"
+	}
+	return false
+}
```

Comment:
For non-OAuth destinations, we would not have `auth.type` in destination definition. Hence it can be a valid-case, we don't need to handle error here as it is expected.

Reply:
What about any other error we can get while making a call to `NestedMapLookup` other then the keys are not present?

Reply:
Are we supposed to ignore those?

Reply:
The purpose of this function is to find if the destination is an oauth destination or not. Until it is serving that purpose we can ignore other errors.